### PR TITLE
fix #278 Handle the send message event on Enter key press

### LINKED ISSUE #278: Handle the send message event on Enter key press
When we use to type the message using Japanese text then we get the suggestion dropdown. If we select the option from the dropdown by pressing the enter key then the "send-message" events get triggered. We need some functionality to handle this scenario.

### PR DESCRIPTION
**Pull request by itTkm**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature #278 
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

I have tested in my local Mac (Monterey 12.3.1) with bellow browsers:
* Chrome 100.0.4896.88（Official Build） （arm64）
* Edge 101.0.1210.39 (Official Build) (arm64)
* Safari 15.4 (17613.1.17.1.13)

### CODE REVIEW

File: src/lib/Room/RoomFooter/RoomFooter.vue [hunk 1]
```diff
@@ -333,7 +333,13 @@ export default {
...
 
 		this.getTextareaRef().addEventListener('keyup', e => {
-			if (e.key === 'Enter' && !e.shiftKey && !this.fileDialog) {
+			if (
+				e.key === 'Enter' &&
+				e.isComposing &&
+				!e.shiftKey &&
+				!this.fileDialog
+			) {
+				console.log('keyup', e.isComposing)
```
Comment (antoine92190) [inline src/lib/Room/RoomFooter/RoomFooter.vue:342]: Could you remove the `console.log`?

**Review by antoine92190 — CHANGES_REQUESTED** [review]

**Review by antoine92190 — APPROVED** [review]

**Comment by antoine92190** [comment]
I'd like to have @thu-san feedback before merging this. @thu-san could you test this PR and let us know if it's working on your side as well?

**Comment by itTkm** [comment]
Hi @thu-san @antoine92190 
Thank you for your feedback @thu-san .
I'm also reproduce that issue in my environment.

I have fixed with 0dc88f5.
Could you please confirm this changes?

https://user-images.githubusercontent.com/61522301/168406838-84d7e90d-e20d-4de9-9f06-5a77d0186292.mov

**Comment by antoine92190** [comment]
@thu-san Could you please test this on your side when you have some time?

**Comment by thu-san** [comment]
@itTkm @antoine92190 
sorry for late reply.
I have checked and it's working fine now!

**Comment by natsumework** [comment]
@antoine92190  Version 1.5.6 doesn't seem to contain this fix? Could you please release this fix?

**Comment by antoine92190** [comment]
Are you using next branch?

**Comment by natsumework** [comment]
No, I am using master branch. And my package.json is like this.

```
"dependencies": {
    "vue-advanced-chat": "1.5.6"
}
```

I checked the node_modules/vue-advanced-chat folder and made sure it was upgraded to version 1.5.6.

image

Also, I confirmed the source code, but it doesn't seem to contain this fix.

image

**Comment by antoine92190** [comment]
Could you try the new version `1.5.7`?

**Comment by natsumework** [comment]
@antoine92190 

I'm trying to figure out why some situations are still triggering the "send-message" event accidentally.
According to this reference:  https://github.com/w3c/uievents/issues/202#issuecomment-411238163

The event order will be `composition-start` / `input` / `compotision-end` / `keyup` on Mac.

This order may result in the following situation:

```
isComposed = true
{event: “compositionstart”,  data: “”, isComposing: true}
{event: “input”, data: “z”, isComposing: tru}
{event: “compositionend”, “data: “z”, isComposing: false}
{event: “keyup”, “key”: “z”}
isComposed = true
{event: “keyup”, “key”: “Enter”} // Will send message here
isComposed = true
```

So  it is still possible to accidentally trigger the "send-message" event.

---

It seems to have a different order on windows, so I'm still researching how to solve this problem.

**Pull request merged**